gandalf.js

```diff
@@ -2,81 +2,75 @@
  * Extras Plugin
```

Comment:
SCRIPT-Consider extracting this repeated logic into a separate function to avoid duplication.